src/mame/amstrad/pda600.cpp

```diff
@@ -206,29 +411,161 @@ static GFXDECODE_START( gfx_pda600 )
 GFXDECODE_END
 
 
+TIMER_DEVICE_CALLBACK_MEMBER(pda600_state::pen_update_timer)
+{
+	u8 pen = m_pen[0]->read();
+
+	// Reduce the update rate when the pen is not held down
+	if (!pen && ++m_pen_cnt < 20)
+		return;
+	else
+		m_pen_cnt = 0;
+
+	m_pen_data[0] = 0x80;               // Start of new data
+	m_pen_data[0] |= pen << 6;          // Pen up/down
+	m_pen_data[0] |= m_rtc_irq << 5;    // RTC IRQ status
+	m_pen_data[0] |= m_battery->read(); // Battery status
+
+	// Pen position (updated only when the pen is down)
+	if (pen)
+	{
+		u16 penx = m_pen[1]->read();
+		u16 peny = m_pen[2]->read();
+		m_pen_data[1] = penx & 0x7f;
+		m_pen_data[2] = (penx >> 7) & 0x7f;
+		m_pen_data[3] = peny & 0x7f;
+		m_pen_data[4] = (peny >> 7) & 0x7f;
+	}
+
+	// Data checksum
+	m_pen_data[5] = m_pen_data[0] + m_pen_data[1] + m_pen_data[2] + m_pen_data[3] + m_pen_data[4];
+
+	// Start CSIO clock
+	m_pen_shift = 0;
+	m_csio_clk_timer->adjust(attotime::zero);
+}
+
+
+TIMER_CALLBACK_MEMBER(pda600_state::csio_clk_timer)
+{
+	if (m_pen_shift < 48)
+	{
+		m_maincpu->cks_w(0);
+		m_maincpu->rxs_cts1_w(BIT(m_pen_data[m_pen_shift / 8], m_pen_shift & 7));
+		m_maincpu->cks_w(1);
+		m_pen_shift++;
```

Comment:
It would be nice to get correct duty cycle for this clock.  Although we don’t typically implement limits, the Z180 CSIO requires the external clock to be held low for a minimum of 11.5 and held high for a minimum of 5 clock cycles.  An instantaneous impulse like this wouldn’t work in real life.

Reply:
At one point I wanted to do this, but I decided against it because I have no hw info and there is no point in using a guessed value, at least the way it is now it's clear that it's not accurate :-)